includes/Checker/Checks/Trademarks_Check.php

```diff
@@ -0,0 +1,543 @@
+		$potential_readme_files = array_filter(
+			$readme_list,
+			function ( $file ) use ( $plugin_relative_path ) {
+				$file = str_replace( $plugin_relative_path, '', $file );
+				if ( ! strpos( $file, '/' ) ) {
+					return true;
+				}
+			}
+		);
+
+		// Find the .txt versions of the readme files.
+		$readme_txt = array_filter(
+			$potential_readme_files,
+			function ( $file ) {
+				return preg_match( '/^readme\.txt$/i', basename( $file ) );
+			}
+		);
+
+		// Find the .md versions of the readme files.
+		$readme_md = array_filter(
+			$potential_readme_files,
+			function ( $file ) {
+				return preg_match( '/^readme\.md$/i', basename( $file ) );
+			}
+		);
+
+		// If there's a .txt version, ignore .md versions.
+		$readme = ( ! empty( $readme_txt ) ) ? $readme_txt : $readme_md;
+
+		// If the readme file does not exist, then skip test.
+		if ( empty( $readme ) ) {
+			return;
+		}
+
+		$matches = array();
+		// Get the plugin name from readme file.
+		$file = self::file_preg_match( '/===(.*)===/i', $files, $matches );
+
+		if ( ! $file ) {
+			return;
+		}
+
+		$name = isset( $matches[1] ) ? $matches[1] : '';
+
+		try {
+			$this->validate_name_has_no_trademarks( $name );
+		} catch ( Exception $e ) {
+			$this->add_result_error_for_file( $result, $file, $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Checks the readme file for default text.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 */
+	private function check_for_name_in_main_file( Check_Result $result ) {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_main_file = WP_PLUGIN_DIR . '/' . $result->plugin()->basename();
+		$plugin_header    = get_plugin_data( $plugin_main_file );
+
+		if ( ! empty( $plugin_header['Name'] ) ) {
+			try {
+				$this->validate_name_has_no_trademarks( $plugin_header['Name'] );
+			} catch ( Exception $e ) {
+				$this->add_result_error_for_file( $result, $plugin_main_file, $e->getMessage() );
+			}
+		}
+	}
+
+	/**
+	 * Checks the readme file for default text.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 */
+	private function check_for_slug( Check_Result $result ) {
+		$plugin_slug = basename( $result->plugin()->path() );
+
+		try {
+			$this->validate_slug_has_no_trademarks( $plugin_slug );
+		} catch ( Exception $e ) {
+			$this->add_result_error_for_file( $result, WP_PLUGIN_DIR . '/' . $result->plugin()->basename(), $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Determines if we find a trademarked term in plugin name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_name The plugin name.
+	 *
+	 * @throws Exception Thrown if we found trademarked term in plugin name.
+	 */
+	private function validate_name_has_no_trademarks( $plugin_name ) {
+		if ( empty( $plugin_name ) ) {
+			return;
+		}
+
+		$check = $this->has_trademarked_slug( $plugin_name );
```

Comment:
See my other comment below:
```suggestion
		$check = $this->has_trademarked_slug( sanitize_title_with_dashes( $plugin_name ) );
```